phobos/blender/operators/editing.py

```diff
@@ -2993,78 +3000,101 @@ def execute(self, context):
         Returns:
 
         """
+        self.executeMessage = []
         self.joints = self.isLinearChain(
             [obj for obj in bpy.context.selected_objects if obj.phobostype == 'link']
         )
         # prepare data used in both cases
         roots = [link for link in self.joints if link.parent not in self.joints]
         if len(roots) != 1:
-            log("Selected joints are not all connected.", 'WARNING')
+            self.executeMessage.append("Careful, the selected joints are not all connected")
             if self.linear_chain:
-                return {'CANCELLED'}
+                return {'FINISHED'}
+
         if self.mechanism_name:
+            for ob in context.scene.objects:
+                if ob.phobostype == "submechanism":
+                    if ob.name == self.mechanism_name:
+                        self.executeMessage.append("Another mechanism with this name already exists")
+                        return {'FINISHED'}
             if self.linear_chain:
-                root = roots[0]
-                root['submechanism/type'] = '{0}R'.format(len(self.joints))
-                root['submechanism/spanningtree'] = list(self.joints)
-                root['submechanism/active'] = list(self.joints)
-                root['submechanism/independent'] = list(self.joints)
-                for i in range(len(self.joints)):
-                    self.joints[i]['submechanism/jointname'] = str(i + 1)
-            else:
-                root, freeloader_joints = eUtils.getNearestCommonParent(self.joints)
+                base = roots[0]
+                parameters = {
+                    'type': '{0}R'.format(len(self.joints)),
+                    #'jointnames': [j["joint/name"] for j in self.joints], #Is autogenerated
+                    'jointnames_spanningtree': [j["joint/name"] for j in self.joints],
+                    'jointnames_active': [j["joint/name"] for j in self.joints],
+                    'jointnames_independent': [j["joint/name"] for j in self.joints],
+                    'name': self.mechanism_name,
+                    'contextual_name': self.mechanism_name
```

Comment:
We should separate name and contextual_name.
Name defines a general name for this submechanism and contextual_name for this specific submech.
Imagine having  a symmetric robot that has a parallelogram chain for the shoulder.
Then e.g. shoulder_parallelogram would be the name of both, and e.g. contextual_names would be left_shoulder and right_shoulder